### PR TITLE
Add azure_info_tags into the doc_fragments

### PR DESCRIPTION
**Pull request by mmyyrroonn**
use azure_info_tags in the azure_rm_containerregistry_facts

##### SUMMARY

Add a new docfragment azure_info_tags
Modify the tags part of the documentation of the **azure_rm_containerregistry_facts** module
Supports setting ```supports_tags=True``` in the info modules

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

azure_info_tags
azure_rm_containerregistry_facts

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by MyronFanQiu** [comment]
@zikalino Hi, Zim! I opened this PR based on our discussion. But I'm not pretty familiar with the doc fragment part and I may miss something. Please help me check this PR when you are free. Thanks!

**Comment by zikalino** [comment]
@MyronFanQiu looks good, we could also add support for **support_tags=True** for _info modules in the same PR?

**Comment by zikalino** [comment]
we could have sth similar in azure_rm_common:

AZURE_TAG_ARGS = dict(
    tags=dict(type='dict'),
    append_tags=dict(type='bool', default=True),
)

--->

AZURE_TAG_ARGS_INFO = dict(
            tags=dict(type='list')
)

**Comment by MyronFanQiu** [comment]
@zikalino ok~ Let me have a try.

**Comment by MyronFanQiu** [comment]
@zikalino It supports ```supports_tags=True``` in the info modules now, and I add a weak restriction onto the function ```validate_tags``` for info modules. If you have any comment or suggestion, please let me know. Thanks!

**Comment by Fred-sun** [comment]
@zikalino Please review this PR when you're available! Thanks!

**Comment by MyronFanQiu** [comment]
@yuwzho @yungezz Hi! When you have time, please take a look at this PR~

**Comment by Fred-sun** [comment]
@yungezz @yuwzho  Please help review this PR when you're available! Thanks!

File: lib/ansible/module_utils/azure_rm_common.py [hunk 1]
```diff
@@ -379,6 +386,9 @@ def validate_tags(self, tags):
             for key, value in tags.items():
                 if not isinstance(value, str):
                     self.fail("Tags values must be strings. Found {0}:{1}".format(str(key), str(value)))
+        else:
+            if not isinstance(tags, list) and not isinstance(tags, dict):
+                self.fail("Tags must be a dictionary of string:string values or list of string values in info modules")
```
Comment (yuwzho) [inline lib/ansible/module_utils/azure_rm_common.py:484]: does it need to check value when the tag is dict?

Comment (mmyyrroonn) [reply lib/ansible/module_utils/azure_rm_common.py:484]: From my view, the ideal solution is to make sure that ```tags``` is the list type in the facts module. However, there are already many facts modules and I'm not sure that all of them supports the list type. A soft check(supports list and dict) may avoid possible errors. For the new module, if we use ```AZURE_INFO_TAG_ARGS```, the type of tags will be limited to list.

**Comment by Fred-sun** [comment]
@yungezz @yuwzho @zikalino Please help take a review this PR and push for merged when you're free! Thank you very much!

**Comment by Fred-sun** [comment]
@yungezz @zikalino review_need

**Comment by Fred-sun** [comment]
@yungezz @zikalino @yuwzho Please help review this PR when you're free, and push for merged. Thanks a lot!

**Review by samccann — APPROVED** [review]
docs portion lgtm

**Comment by Fred-sun** [comment]
@yungezz @MyronFanQiu Maybe we can merge this PR? Please confirm, Thanks a lot!

**Comment by Fred-sun** [comment]
@yungezz @zikalino This PR has been approved for a long time. Please help to review and promote the merger. Thank you!

**Comment by Fred-sun** [comment]
@MyronFanQiu Please rebase the PR when you're free! Thanks a lot!

**Comment by Fred-sun** [comment]
@MyronFanQiu Would you please help rebase the PR? Thank you very much!

**Comment by Fred-sun** [comment]
kindly ping!

**Comment by Fred-sun** [comment]
@MyronFanQiu

**Comment by Fred-sun** [comment]
ready_for_review

**Pull request closed without merging**